src/category_theory/sites/sheaf.lean

```diff
@@ -54,13 +54,125 @@ variables (J : grothendieck_topology C)
 -- We follow https://stacks.math.columbia.edu/tag/00VL definition 00VR
 
 /--
-A sheaf of A is a presheaf P : C^op => A such that for every X : A, the
-presheaf of types given by sending U : C to Hom_{A}(X, P U) is a sheaf of types.
+A sheaf of A is a presheaf P : Cᵒᵖ => A such that for every E : A, the
+presheaf of types given by sending U : C to Hom_{A}(E, P U) is a sheaf of types.
 
 https://stacks.math.columbia.edu/tag/00VR
 -/
 def is_sheaf (P : Cᵒᵖ ⥤ A) : Prop :=
-∀ X : A, presieve.is_sheaf J (P ⋙ coyoneda.obj (op X))
+∀ E : A, presieve.is_sheaf J (P ⋙ coyoneda.obj (op E))
+
+section limit_sheaf_condition
+open presieve presieve.family_of_elements limits
+variables (P : Cᵒᵖ ⥤ A) {X : C} (S : sieve X) (R : presieve X) (E : Aᵒᵖ)
+
+/-- Given a sieve `S` on `X : C`, a presheaf `P : Cᵒᵖ ⥤ A`, and an object `E` of `A`,
+    the cones over the natural diagram `S.arrows.diagram.op ⋙ P` associated to `S` and `P`
+    with cone point `E` are in 1-1 correspondence with sieve_compatible family of elements
+    for the sieve `S` and the presheaf of types `Hom (E, P -)`. -/
+@[simps] def cones_equiv_sieve_compatible_family :
+  (functor.cones (S.arrows.diagram.op ⋙ P)).obj E ≃
```

Comment:
Can you use dot notation for `functor.cones`?

Reply:
Indeed I can! I didn't realize that.